modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -195,16 +211,47 @@ public TableManager(
     /** {@inheritDoc} */
     @Override
     public void start() {
+        tablesCfg.tables().any().replicas().listen(replicasCtx -> {
+            if (!busyLock.enterBusy()) {
```

Comment:
Let's extract it to explicit onReplicasChange() like it's done with onTableCreate(), onTableDelete(), etc. WDYT?